olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -769,6 +894,77 @@ client::CancellationToken AuthenticationClient::Impl::SignOut(
   });
 }
 
+client::CancellationToken AuthenticationClient::Impl::IntrospectApp(
+    std::string access_token, IntrospectAppCallback callback) {
+  auto introspect_app_task = [=](client::CancellationContext context)
+      -> client::ApiResponse<IntrospectAppResult, client::ApiError> {
+    if (!settings_.network_request_handler) {
+      client::ApiError result({static_cast<int>(http::ErrorCode::IO_ERROR),
+                               "Cannot introspect app while offline"});
+      return std::move(result);
+    }
+    std::string url = settings_.token_endpoint_url + kIntrospectAppEndpoint;
+    auto network_settings = http::NetworkSettings().WithProxySettings(
+        settings_.network_proxy_settings.value_or(
+            http::NetworkProxySettings{}));
+    auto request =
+        http::NetworkRequest(url)
+            .WithVerb(http::NetworkRequest::HttpVerb::GET)
+            .WithHeader(http::kAuthorizationHeader,
+                        generateBearerHeader(access_token))
+            .WithHeader(http::kUserAgentHeader, http::kOlpSdkUserAgent)
+            .WithSettings(std::move(network_settings));
+
+    auto payload = std::make_shared<std::stringstream>();
+    auto send_outcome = settings_.network_request_handler->Send(
+        request, payload,
+        [callback, payload](const http::NetworkResponse& network_response) {
+          auto response_status = network_response.GetStatus();
+          auto error_msg = network_response.GetError();
+
+          if (response_status != 200) {
+            // Network error response not available
+            AuthenticationError result(response_status, error_msg);
+            callback(std::move(result));
+            return;
+          }
+
+          auto document = std::make_shared<rapidjson::Document>();
```

Comment:
I wonder, do we still need to use shared_ptr for this? It is not passed anymore to the result class.